### PR TITLE
ARROW-10917: [Doc] Update feature matrix for Rust

### PR DESCRIPTION
**Pull request by nevi-me**
Updates matrix for work done since `2.0.0`

### CODE REVIEW

File: docs/source/status.rst [hunk 1]
```diff
@@ -169,18 +170,18 @@ Notes:
...
-+=============================+=======+========+=======+
-| Schema export               | ✓     | ✓      | ✓     |
-+-----------------------------+-------+--------+-------+
-| Array export                | ✓     | ✓      | ✓     |
-+-----------------------------+-------+--------+-------+
-| Schema import               | ✓     | ✓      | ✓     |
-+-----------------------------+-------+--------+-------+
-| Array import                | ✓     | ✓      | ✓     |
-+-----------------------------+-------+--------+-------+
++-----------------------------+-------+--------+-------+-------+
```
Comment (nevi-me) [inline docs/source/status.rst:173]: @jorgecarleitao this one's your baby ❤️, may you please populate what's supported on the C Data interface in Rust.
Please push directly on my branch

File: docs/source/status.rst [hunk 2]
```diff
@@ -161,6 +161,7 @@ Notes:
 
 * \(1) No support for handshake or DoExchange.
 * \(2) Support using AspNetCore authentication handlers.
+* \(3) Flight integration testing for Rust is still ongoing as of version 3.0.0, see ARROW-8853.
```
Comment (nevi-me) [inline docs/source/status.rst:164]: @carols10cents I haven't ticked anything here, as we haven't merged the flight test PR

**Comment by mqy** [comment]
The custom metadata is not implemented for in file footer, dictionary and record batch, should we add this into 3.0.0?

File: docs/source/status.rst (hunk 2 above)
Comment (pitrou) [reply docs/source/status.rst:164]: I don't think we should add information about work-in-progress features here. This page is for end-users.

Comment (nevi-me) [reply docs/source/status.rst:164]: Thanks, I'll remove WIP

**Comment by nevi-me** [comment]
> The custom metadata is not implemented for file footer, dictionary and record batch, should we add this into 3.0.0?

Yes please, I thought it was implemented. Or do you mean in the IPC format? I'll untick it in the meantime

**Comment by mqy** [comment]
> > The custom metadata is not implemented for file footer, dictionary and record batch, should we add this into 3.0.0?
> 
> Yes please, I thought it was implemented. Or do you mean in the IPC format? I'll untick it in the meantime
I mean the IPC format, I'll push some incomplete changes soon.

**Review by alamb — COMMENTED** [review]
This really cool to see all the work for Rust over the last few months!

**Comment by alamb** [comment]
@nevi-me  is this ready to go? I think the 3.0 release is just about to happen so we should probably merge it in soon (TM)

**Comment by pitrou** [comment]
I've taken the liberty of updating this PR. Will merge if CI green.

**Review by pitrou — APPROVED** [review]

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @pitrou

**Comment by pitrou** [comment]
Green CI on my fork: https://github.com/pitrou/arrow/actions/runs/480864596

**Pull request closed without merging**